apps/channels/forms.py

```diff
@@ -71,6 +71,33 @@ def get_success_message(self, channel: ExperimentChannel):
         return f"Use the following URL when setting up the webhook: {channel.webhook_url}"
 
 
+class InAppChannelForm(ExtraFormBase):
+    client_id = forms.CharField(label="Client ID", max_length=100)
```

Comment:
Is this `client_id` different from the one on the `MessagingProvider`. If they are the same the it doesn't seem useful to have it here as well.

Reply:
Can you confirm this understanding:

* `MessagingProvider.client_id`: this is an Oauth client ID used to authenticate with SureAdhere
* `Channel.client_id`: this is a workspace ID in SureAdhere (is a workspace SureAdhere's 'tenent'?)

Is `Client ID` the right name for it then?

It's also not clear to me what this is used for since it isn't included in the requests to send messages so it appears to only be used for the webhook.

I think that leads to the question of how the webhooks in SA are configured. Is there documentation on that somewhere? How does SA know which webhook to call for a specific message?

Reply:
Typically webhooks are configured in the sending system so that it know where to send messages. There are 2 options:

1. A generic webhook where the request body contains some information that identifies where the message is coming from e.g. client_id
2. A webhook that contains some identifying info in it e.g. experiment_id or client_id

Either of these approaches are fine but both require some way to configure them in SA.

For now we can revert to using `client_id` since that's what allows you to test but I'd like to hear what the product team has to say.